### PR TITLE
Add JDK9 constructors to `j.m.BigInteger`

### PR DESCRIPTION
**Pull request by armanbilge**
@plokhotnyuk uses these in jsoniter-scala.

https://github.com/plokhotnyuk/jsoniter-scala

### CODE REVIEW

File: unit-tests/shared/src/test/scala/javalib/math/BigIntegerTest.scala [hunk 1]
```diff
@@ -175,4 +176,149 @@ class BigIntegerTest {
 
     assertFalse(jbi1 == jbi2)
   }
+
+  // ported from Scala.js commit cbf86bb dated Oct 23 2020
```
Comment (armanbilge) [inline unit-tests/shared/src/test/scala/javalib/math/BigIntegerTest.scala:180]: @ekrich you mean this comment?

Comment (ekrich) [reply unit-tests/shared/src/test/scala/javalib/math/BigIntegerTest.scala:180]: I didn't see that down there. I guess we had always put them on the top based on the latest port. I had to put some comments in some of the code I ported as there was some Scala Native specific code so it wasn't a straight port. Added for JDK 9+ might be helpful there.

Comment (armanbilge) [reply unit-tests/shared/src/test/scala/javalib/math/BigIntegerTest.scala:180]: I added it to make it clear where the Scala.js ported test came from.

> Added for JDK 9+ might be helpful there.

I wrote those tests myself. Scala.js does not support these APIs and thus has no tests for them.

Comment (ekrich) [reply unit-tests/shared/src/test/scala/javalib/math/BigIntegerTest.scala:180]: Sounds good, thanks. It seems sometime it should be cut and dry but is messier in practice.

**Review by WojciechMazur — APPROVED** [review]

**Pull request merged**